sdks/python/apache_beam/transforms/userstate_test.py

```diff
@@ -415,6 +422,82 @@ def clear_values(self, bag_state=beam.DoFn.StateParam(BAG_STATE)):
         ['extra'],
         StatefulDoFnOnDirectRunnerTest.all_records)
 
+  def test_simple_set_stateful_dofn(self):
+    class SimpleTestSetStatefulDoFn(DoFn):
+      BUFFER_STATE = SetStateSpec('buffer', BytesCoder())
+      EXPIRY_TIMER = TimerSpec('expiry', TimeDomain.WATERMARK)
+
+      def process(self, element, buffer=DoFn.StateParam(BUFFER_STATE),
+                  timer1=DoFn.TimerParam(EXPIRY_TIMER)):
+        unused_key, value = element
+        buffer.add(str(value).encode('latin1'))
+        timer1.set(20)
+
+      @on_timer(EXPIRY_TIMER)
+      def expiry_callback(self, buffer=DoFn.StateParam(BUFFER_STATE)):
+        yield b''.join(sorted(buffer.read()))
+
+    with TestPipeline() as p:
+      test_stream = (TestStream()
+                     .advance_watermark_to(10)
+                     .add_elements([1, 2, 3])
+                     .add_elements([2])
+                     .advance_watermark_to(24))
+      (p
+       | test_stream
+       | beam.Map(lambda x: ('mykey', x))
+       | beam.ParDo(SimpleTestSetStatefulDoFn())
+       | beam.ParDo(self.record_dofn()))
```

Comment:
Is there a reason assert_that doesn't work here?

Reply:
I haven't tried the `assert_that` here. I took reference from already existing test cases. I will update the test cases.

Reply:
`assert_that` doesn't work in case of state and timer. Here is the jira ticket filed long back: https://issues.apache.org/jira/browse/BEAM-5295 I can take care of this in a separate ticket.